lib/devices/phone_device_manager.js

```diff
@@ -0,0 +1,78 @@
+import assert from 'assert';
+import Debug from 'debug';
+
+import Testdroid from 'testdroid-client';
+
+let debug = Debug('taskcluster-docker-worker:devices:phoneManager');
+
+export default class PhoneDeviceManager {
+  constructor(config) {
+    assert(config.deviceManagement.phone.sims, "Must supply the number of sims required");
+    assert(config.deviceManagement.phone.type, "Must supply the type of phone");
+    assert(config.testdroid, "Must supply testdroid configuration");
+    assert(config.testdroid.url, "Must supply testdroid cloud url");
+    assert(config.testdroid.username, "Must supply testdroid cloud username");
+    assert(config.testdroid.password, "Must supply testdroid cloud password");
+
+    this.config = config;
+    this.client = new Testdroid(
+        config.testdroid.url,
+        config.testdroid.username,
+        config.testdroid.password
+    );
+    this.deviceFilter = {
+      'type': config.deviceManagement.phone.type,
+      'sims': config.deviceManagement.phone.sims
+    };
+  }
+
+  async devices() {
+    let devices = await this.client.getDevices(this.deviceFilter);
+    let deviceList = devices.map((device) => {
+      return new Phone(device);
+    });
+
+    debug(`List of ${deviceList.length} phones created`)
+    return deviceList;
+  }
```

Comment:
So ideally this list of devices will be devices the worker currently has sessions for. It will create the sessions and then base claim capacity on how many phones for sure it has for tasks. 

Something better could be done here.  Right now it's only used for getting a list of online devices and once bitbar finishes some Api work we'll be able to create sessions before claiming. That part I'm still waiting on from them.